public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -167,14 +167,20 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   const isCloudAlertRuleType = isCloudAlertingRuleByType(type);
   const [showResetModeModal, setShowResetModal] = useState(false);
 
+  const simplifiedQueryInForm = editorSettings?.simplifiedQueryEditor;
+
   const { isAdvancedMode, simpleCondition, setSimpleCondition } = useAdvancedMode(
-    editorSettings,
+    simplifiedQueryInForm,
     isGrafanaAlertingType,
     isNewFromQueryParams,
     dataQueries,
     expressionQueries
   );
 
+  useEffect(() => {
```

Comment:
this syncs  `editorSettings.simplifiedQueryEditor` in the form with `isAdvancedmode`

Reply:
I've just pushed a commit with this simplification